x-pack/plugins/ml/public/alerting/anomaly_detection_alerts_table/alert_actions.tsx

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButtonIcon,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiPopover,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import React, { useCallback, useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { CaseAttachmentsWithoutOwner } from '@kbn/cases-plugin/public';
+import { AttachmentType } from '@kbn/cases-plugin/common';
+import { EcsSecurityExtension as Ecs } from '@kbn/securitysolution-ecs';
+import {
+  ALERT_RULE_NAME,
+  ALERT_RULE_UUID,
+  ALERT_STATUS,
+  ALERT_STATUS_ACTIVE,
+  ALERT_UUID,
+} from '@kbn/rule-data-utils';
+import { useBulkUntrackAlerts } from '@kbn/triggers-actions-ui-plugin/public';
+import { type Alert } from '@kbn/triggers-actions-ui-plugin/public/types';
+import { PLUGIN_ID } from '../../../common/constants/app';
+import { useMlKibana } from '../../application/contexts/kibana';
+
+export interface AlertActionsProps {
+  alert: Alert;
+  ecsData: Ecs;
+  id?: string;
+  refresh: () => void;
+  setFlyoutAlert: React.Dispatch<React.SetStateAction<any | undefined>>;
+}
+
+export function AlertActions({
+  alert,
+  ecsData,
+  id: pageId,
+  refresh,
+  setFlyoutAlert,
+}: AlertActionsProps) {
+  const {
+    cases,
+    http: {
+      basePath: { prepend },
+    },
+  } = useMlKibana().services;
+  const casesPrivileges = cases?.helpers.canUseCases();
+
+  const { mutateAsync: untrackAlerts } = useBulkUntrackAlerts();
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+
+  const ruleId = alert[ALERT_RULE_UUID]?.[0] ?? null;
+  const alertId = alert[ALERT_UUID]?.[0] ?? '';
+
+  const linkToRule = ruleId
+    ? prepend(`/app/management/insightsAndAlerting/triggersActions/rule/${ruleId}`)
+    : null;
+
+  const caseAttachments: CaseAttachmentsWithoutOwner = useMemo(() => {
+    return ecsData?._id
+      ? [
+          {
+            alertId: alertId ?? '',
+            index: ecsData?._index ?? '',
+            type: AttachmentType.alert,
+            rule: {
+              id: ruleId,
+              name: alert[ALERT_RULE_NAME]![0],
+            },
+            owner: PLUGIN_ID,
+          },
+        ]
+      : [];
+  }, [alert, alertId, ecsData?._id, ecsData?._index, ruleId]);
+
+  const isActiveAlert = useMemo(() => alert[ALERT_STATUS]![0] === ALERT_STATUS_ACTIVE, [alert]);
+
+  const onSuccess = useCallback(() => {
+    refresh();
+  }, [refresh]);
+
+  const createCaseFlyout = cases!.hooks.useCasesAddToNewCaseFlyout({ onSuccess });
+  const selectCaseModal = cases!.hooks.useCasesAddToExistingCaseModal({ onSuccess });
+
+  const closeActionsPopover = () => {
```

Comment:
The alerts popover is staying open on different user actions - for example opening the job selection flyout:

<img width="820" alt="image" src="https://github.com/elastic/kibana/assets/7405507/282321dc-ac69-48fd-a930-9182bedcc7ce">

Or editing the swim lane severity slider:

<img width="821" alt="image" src="https://github.com/elastic/kibana/assets/7405507/a9c1dd93-59fa-4222-a03e-69983e735683">

Reply:
Fixed in bcbbecd2dc5a9e52902b77d973c9ea46e29e8f84